LibraBFT/Impl/Consensus/BlockStorage/Properties/BlockStore.agda

```diff
@@ -69,27 +70,38 @@ module executeAndInsertBlockESpec (bs0 : BlockStore) (block : Block) where
   record ContractOk (bs' : BlockStore) (eb : ExecutedBlock) : Set where
     constructor mkContractOk
     field
-      ebBlockData≡ : eb ^∙ ebBlock ≡L block at bBlockData
-      bsInv        : ∀ pre → pre ^∙ lBlockStore ≡ bs0
-                     → Preserves BlockStoreInv pre (pre & lBlockStore ∙~ bs')
-      qcPost       : QCProps.∈Post⇒∈PreOrBT (_≡ block ^∙ bQuorumCert) (bs' ^∙ bsInner) (bs0 ^∙ bsInner)
+      ebBlock≈ : eb ^∙ ebBlock ≈Block block
+      bsInv    : ∀ pre → pre ^∙ lBlockStore ≡ bs0
+                 → Preserves BlockStoreInv pre (pre & lBlockStore ∙~ bs')
+      -- TODO-2: The fields below should be about blocks, not QCs. The property
+      -- for QCs then follows as a consequence
+      qcPost : ∀ qc → qc QCProps.∈BlockTree (bs' ^∙ bsInner)
```

Comment:
As currently stated, this could use `∈Post⇒∈PreOrBT`.  I was thinking about generalising it to allow more types than just `QuorumCert`, which would enable the change to make this about `Block`s too.